k9mail/src/test/java/com/fsck/k9/view/MessageHeaderTest.java

```diff
@@ -0,0 +1,56 @@
+package com.fsck.k9.view;
+
+import android.util.AttributeSet;
+
+import com.fsck.k9.K9RobolectricTestRunner;
+import com.fsck.k9.mail.Address;
+import com.fsck.k9.mail.Message;
+import com.fsck.k9.mail.internet.MimeMessage;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RuntimeEnvironment;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+
+@RunWith(K9RobolectricTestRunner.class)
+public class MessageHeaderTest {
```

Comment:
You should try to avoid testing View classes. The best way to achieve this is by extracting logic to separate classes.

In this case it would probably involve extracting much of the existing logic, which is quite some work on its own. The next best thing is making `shouldShowSender()` static. That way it can be tested without having to instantiate `MessageHeader`.